internal/provider/resource_docker_service.go

```diff
@@ -83,10 +79,9 @@ func resourceDockerService() *schema.Resource {
 							Elem: &schema.Resource{
 								Schema: map[string]*schema.Schema{
 									"image": {
-										Type:             schema.TypeString,
-										Description:      "The image name to use for the containers of the service",
-										Required:         true,
-										DiffSuppressFunc: suppressIfSHAwasAdded(),
+										Type:        schema.TypeString,
+										Description: "The image name to use for the containers of the service. Use the `docker_image` resource for this, as shown in the examples. Although direct image names like `nginx:latest` works it is not recommend to trigger updates",
```

Comment:
Should be able to use `docker_image` or `docker_registry_image` (from resource or data source) as long as we can get a full "repo_digest" string from each of them.

(Local `docker_image`s that haven't been pushed to a repository, will require that same image to exist on all swarm nodes that will run the service.)

Reply:
I adapted the description in 4d0f69d7acffb77f88e450d39bb002e2b52633a5 and tested the cases locally